common/persistence/sql/workflowStateMaps.go

```diff
@@ -374,25 +354,17 @@ func updateRequestCancelInfos(
 			return serviceerror.NewInternal(fmt.Sprintf("Failed to update request cancel info. Failed to execute update query. Error: %v", err))
 		}
 	}
-	if deleteInfo == nil {
-		return nil
-	}
-	result, err := tx.DeleteFromRequestCancelInfoMaps(sqlplugin.RequestCancelInfoMapsDeleteFilter{
-		ShardID:     shardID,
-		NamespaceID: namespaceID,
-		WorkflowID:  workflowID,
-		RunID:       runID,
-		InitiatedID: deleteInfo,
-	})
-	if err != nil {
-		return serviceerror.NewInternal(fmt.Sprintf("Failed to update request cancel info. Failed to execute delete query. Error: %v", err))
-	}
-	rowsAffected, err := result.RowsAffected()
-	if err != nil {
-		return serviceerror.NewInternal(fmt.Sprintf("Failed to update request cancel info. Failed to verify number of rows deleted. Error: %v", err))
-	}
-	if int(rowsAffected) != 1 {
-		return serviceerror.NewInternal(fmt.Sprintf("Failed to update request cancel info. Deleted %v rows instead of 1", rowsAffected))
+
+	for _, deleteInfoID := range deleteInfos {
+		if _, err := tx.DeleteFromRequestCancelInfoMaps(sqlplugin.RequestCancelInfoMapsDeleteFilter{
+			ShardID:     shardID,
+			NamespaceID: namespaceID,
+			WorkflowID:  workflowID,
+			RunID:       runID,
+			InitiatedID: &deleteInfoID,
```

Comment:
probably we don't need this InitiatedID to be a pointer anymore.

Reply:
there are 2 behaviors, one behavior select / delete all, one behavior select / delete one single row
that is why this is a pointer